kernel/host/xmicterm/host/xmicterm.cc

```diff
@@ -129,16 +144,73 @@ class file_logger {
     std::unordered_map<uint16_t, std::unique_ptr<std::ofstream>> files;
 };
 
+/// Wrapper to the micctrl application, that controls the xeon phi coprocessor
+class mic_ctrl {
+public:
+    mic_ctrl(int adapter_)
+        :adapter(adapter_) {
+      }
+
+
+    void boot() {
+      std::stringstream ss;
+      ss << "sudo sh -c \"echo \\\"boot:elf:`pwd`/boot64.elf\\\" > /sys/class/mic/mic"
```

Comment:
You can use "boot:linux:" instead of "boot:elf" although the image is still ELF. This tricks the mpss on Linux side into preparing the virtio and similar things. We just don't yet get the needed pointers on the mythos side because the KNC bootloader doesn't place the pointer to the kernel command line string to a known position. But the data is in the KNC's memory (confirmed by memory dump). Just a matter of time...